### ISSUE TITLE
Censoring images

### ISSUE THREAD

**Issue by OttoAllmendinger** [OttoAllmendinger/gnome-shell-screenshot#35]
From https://github.com/OttoAllmendinger/gnome-shell-imgur/issues/44

----

Do you thought about adding an option to censor pictures before uploading them to imgur, like f.i. shown [here](http://i.imgur.com/q4qwt9k.png)?

This would come very handy.

Thanks for this great extension!

**Comment by FDiskas** [comment]
Oh no - do not add image editor ability. There is plenty of them. Just keep it simple as is now.

**Comment by OttoAllmendinger** [comment]
@FDiskas I agree, I probably won't add it. Maybe I will add an "edit" option in the popup menu though. 

A separate extension to hide or highlight portions of the screen would also be interesting.

**Comment by sojusnik** [comment]
Yes, this request is mainly about quickly hiding or sometimes highlighting some areas of the screenshot, not full image editing capabilities like f.i gimp offers. I also prefer simple and clean programs, but censoring screenshots is essential for an image sharing program, imho.

**Comment by OttoAllmendinger** [comment]
I think some integration with https://github.com/OttoAllmendinger/gnome-shell-screenshot/issues/35 would do the job

also editing post-capture in Gimp would be another option

**Comment by sojusnik** [comment]
The solution with draw on your screen looks solid!

**Comment by real-artswan** [comment]
At the first, thanks for the one of the most worth extension! Great job!

My 2 cents about that. 
I think it must implement a way to be able to 
1. Make an rectangle around arbitrary area of a screenshot
2. Put a text at any place of a screenshot

I would like to help with implementing that but unfortunately I'm not familiar with Gnome API and don't know where to start. Would appreciate links to rtfm.

After all, why do we need such an extension instead of just pressing "PrintScreen" on the keyboard and pasting to the favorite image editor?
I very often need an ability to make few screenshots that describes what need to do, or what is wrong, or what my receiver should think about.

It must be as easy and fast as it is right now, but with an optional extra ability to put frames and texts on top. Running GIMP each time for that is slow, keeping it open all the time... Even don't know.
To ppl who talk about "plenty editors": I don't know any quick tool for Linux (and for Windows btw) that implements such a daily base features, and GIMP is not a solution for a such simple thing

**Comment by OttoAllmendinger** [comment]
Having a basic lightweight image editor is an interesting project, but I'm afraid it just exceeds the scope of this extension. 

Have you tried https://pinta-project.com/pintaproject/pinta/ ?

> I would like to help with implementing that but unfortunately I'm not familiar with Gnome API and don't know where to start. Would appreciate links to rtfm.

The best way is to read existing programs and extensions. Most of the apis are documented here:
https://developer.gnome.org/gtk3/3.0/

> P.S. draw-on-your-screen cant handle popup dialogs which disappear on pressing any button (that's why we have screenshot delay feature)

I think it might be easier to find a fix for that problem than to build a new custom editor.

**Comment by FDiskas** [comment]
good example at: https://github.com/lupoDharkael/flameshot
![](https://github.com/lupoDharkael/flameshot/raw/master/img/preview/animatedUsage.gif)

**Comment by OttoAllmendinger** [comment]
flameshot looks pretty cool! I might just use it as an backend for the extension if it is installed.
